app/assets/javascripts/hikes/hikes_search.js

```diff
@@ -44,14 +46,19 @@ function addDataTable() {
       },
 
       "bDestroy": true,
-      "iDisplayLength": 25,
+      "iDisplayLength": tableDisplayLength,
       "columns": [
         { "width": "50%" },
         { "width": "10%" },
         { "width": "12.5%" },
         { "width": "12.5%" },
         { "width": "15%" }
-      ]
+      ],
+      "fnDrawCallback": function(oSettings) {
+        if ($('#searched_hikes tr').length <= tableDisplayLength) {
```

Comment:
if these selectors aren't expected to change after initial page load then i would suggest caching them in a variable so they aren't executed every time this function is called.